wvdecrypter/cdm/media/cdm/cdm_adapter.h

```diff
@@ -235,7 +237,12 @@ class CdmAdapter : public std::enable_shared_from_this<CdmAdapter>
   std::string cdm_path_;
   std::string cdm_base_path_;
   CdmAdapterClient *client_;
-  std::mutex client_mutex_, decrypt_mutex_;
+  std::mutex client_mutex_;
+  std::mutex decrypt_mutex_;
+  std::mutex m_closeSessionMutex;
+  std::atomic<bool> ms_isCloseSession;
+  std::condition_variable ms_sessionClosingCond;
+  std::vector<std::future<void>> m_futures;
```

Comment:
variable name dont explain his use, could be an idea something like:
```suggestion
  std::vector<std::future<void>> m_asyncTasks;
```